### DISCUSSION TITLE
User roles and authentication

### DISCUSSION THREAD

**Discussion by eaziweb95** [filamentphp/filament#2883] [Help] [answered]
I'd like to add user roles and permissions in my application. The demo application has no roles.
What filament plugin do you advise ?
filament-shield ? filament-access-control ? another ?
Is there any tutorial on this ?

**Comment by danharrin** [comment, marked as answer]
shield is the most popular

**Reply by eaziweb95** [reply]
it requires spatie packages and illuminate contracts ?

**Reply by eaziweb95** [reply]
is there any tutorial on filament application with shield ?

**Reply by ryangjchandler** [reply]
Shield is a third-party plugin for Filament so we don't provide any documentation. You might have some luck here. https://github.com/bezhanSalleh/filament-shield
